module/VuFind/src/VuFind/Controller/MyResearchController.php

```diff
@@ -1766,14 +1783,18 @@ protected function sendRecoveryEmail(UserEntityInterface $user, $config)
                     $config = $this->getConfig();
                     $renderer = $this->getViewRenderer();
                     $method = $this->getAuthManager()->getAuthMethod();
+                    // If target exists create query string to include it as part of reset url
+                    $target = $this->getRequest()->getQuery('target') ? '&target='
+                        . $this->getRequest()->getQuery('target') : null;
```

Comment:
Since you concatenate this to a string down below, I don't think you ever want it to be null -- should probably be empty string instead. Also, it's probably best to encode the value for safety. And it might make sense to refactor to avoid calling getQuery so many times, e.g.:

```
$target = $this->getRequest()->getQuery('target');
$targetParam = $target ? '&target=' . urlencode($target) : '';
```

Then down below, you can use $targetParam in your concatenation, and you can assign $target to $view->target instead of fetching the value yet another time.